OpenRA.Mods.Common/Projectiles/AreaBeam.cs

```diff
@@ -136,7 +139,19 @@ public AreaBeam(AreaBeamInfo info, ProjectileArgs args, Color color)
 			// Update the target position with the range we shoot beyond the target by
 			// I.e. we can deliberately overshoot, so aim for that position
 			var dir = new WVec(0, -1024, 0).Rotate(WRot.FromYaw(towardsTargetFacing));
-			target += dir * info.BeyondTargetRange.Length / 1024;
+			var dist = (args.SourceActor.CenterPosition - target).Length;
+			int extraDist;
+			if (info.MinDistance.Length > dist)
+			{
+				if (info.MinDistance.Length - dist < info.BeyondTargetRange.Length)
+					extraDist = info.BeyondTargetRange.Length;
+				else
+					extraDist = info.MinDistance.Length;
```

Comment:
Won't this add MinDistancw behind the target? Shouldn't it be MinDistance - dist?

Reply:
yes mb, it should be `MinDistance - dist`